plugins/scaffolder/src/components/MultistepJsonForm/ReviewStep.tsx

```diff
@@ -25,38 +25,37 @@ export function getReviewData(
   uiSchemas: UiSchema[],
 ) {
   const reviewData: Record<string, any> = {};
-  for (const key in formData) {
-    if (formData.hasOwnProperty(key)) {
-      const uiSchema = uiSchemas.find(us => us.name === key);
+  const formPropertyNames = new Set(Object.getOwnPropertyNames(formData));
+  const handleSchema = (key: string, uiSchema: UiSchema): void => {
+    let value: any;
+    if (uiSchema['ui:widget'] === 'password') {
+      value = '******';
+    } else {
+      const uiBackstage = uiSchema['ui:backstage'];
+      if (uiBackstage && uiBackstage.review) {
+        const review = uiSchema['ui:backstage'].review as JsonObject;
 
-      if (!uiSchema) {
-        reviewData[key] = formData[key];
-        continue;
+        if (review.mask) {
+          value = review.mask;
+        } else if (review.show) {
+          value = formData[key];
+        }
+      } else {
+        value = formData[key];
       }
-
-      if (uiSchema['ui:widget'] === 'password') {
-        reviewData[key] = '******';
-        continue;
-      }
-
-      if (!uiSchema['ui:backstage'] || !uiSchema['ui:backstage'].review) {
-        reviewData[key] = formData[key];
-        continue;
-      }
-
-      const review = uiSchema['ui:backstage'].review as JsonObject;
-      if (review.mask) {
-        reviewData[key] = review.mask;
-        continue;
-      }
-
-      if (!review.show) {
-        continue;
-      }
-      reviewData[key] = formData[key];
     }
-  }
+    if (value !== undefined) {
+      reviewData[key] = value;
+    }
+    formPropertyNames.delete(key);
+  };
+  uiSchemas
+    .filter(uiSchema => formPropertyNames.has(uiSchema.name))
+    .forEach(uiSchema => handleSchema(uiSchema.name, uiSchema));
```

Comment:
Can we simplify both of these `.forEach`s into one `for` loop instead?

Think there's a little too much indirection and it makes the code quite hard to understand I think now.